### PR TITLE
Fixes #400

### LINKED ISSUE #400: Metaschema DateWithTimezoneDatatype definition allows invalid timezone values
https://github.com/usnistgov/metaschema/blob/a36f579e1e30abb2263895242cdbd2cf4bd29513/schema/xml/metaschema-datatypes.xsd#L38

The timezone portion of the regex is `(Z|[+-][0-9]{2}:[0-9]{2})`

According to RFC 3339, the proper format is:

```
   time-hour       = 2DIGIT  ; 00-23
   time-minute     = 2DIGIT  ; 00-59
  <...>
   time-numoffset  = ("+" / "-") time-hour ":" time-minute
   time-offset     = "Z" / time-numoffset
```

The timezone regex should be `(Z|([+|-])([0-1][0-9]|2[0-3]):([0-5][0-9]))`

There is a similar error here:

https://github.com/usnistgov/metaschema/blob/a36f579e1e30abb2263895242cdbd2cf4bd29513/schema/xml/metaschema-datatypes.xsd#L56

### PR DESCRIPTION
**Pull request by RS-Credentive**
# Committer Notes

Addresses
 #400 

### All Submissions:

- [x] Have you followed the guidelines in our [Contributing](https://github.com/usnistgov/metaschema/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/usnistgov/metaschema/pulls) for the same update/change?
- [x] Have you squashed any non-relevant commits and commit messages? \[[instructions](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History)\]
- [ ] Do all automated CI/CD checks pass?

### Changes to Core Features:

- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you included examples of how to use your new feature(s)?
- [ ] Have you updated all website](https://pages.nist.gov/metaschema) and readme documentation affected by the changes you made? Changes to the website can be made in the website/content directory of your branch.

### CODE REVIEW

**Comment by RS-Credentive** [comment]
I have not updated the documentation on the website, as I believe it is automatically generated from the schema itself.

**Review by aj-stein-nist — APPROVED** [review]
I am but a bit player here, but I like it!

**Review by david-waltermire-nist — CHANGES_REQUESTED** [review]
@RS-Credentive Thanks for the submission. The XML changes look good.

Can you make the same changes to the [JSON data types](https://github.com/usnistgov/metaschema/blob/develop/schema/json/metaschema-datatypes.json)? These need to be kept in sync.

**Comment by RS-Credentive** [comment]
> 

No problem. Thanks for the heads up. I thought they were autogenerated from the metaschema, or I would have already done so.

File: schema/xml/metaschema-datatypes.xsd [hunk 1]
```diff
@@ -26,7 +26,7 @@
 			<xs:documentation>A string representing a 24-hour period with an optional timezone.</xs:documentation>
 		</xs:annotation>
 		<xs:restriction base="xs:date">
-			<xs:pattern value="(((2000|2400|2800|(19|2[0-9](0[48]|[2468][048]|[13579][26])))-02-29)|(((19|2[0-9])[0-9]{2})-02-(0[1-9]|1[0-9]|2[0-8]))|(((19|2[0-9])[0-9]{2})-(0[13578]|10|12)-(0[1-9]|[12][0-9]|3[01]))|(((19|2[0-9])[0-9]{2})-(0[469]|11)-(0[1-9]|[12][0-9]|30)))(Z|[+-][0-9]{2}:[0-9]{2})?"/>
+			<xs:pattern value="(((2000|2400|2800|(19|2[0-9](0[48]|[2468][048]|[13579][26])))-02-29)|(((19|2[0-9])[0-9]{2})-02-(0[1-9]|1[0-9]|2[0-8]))|(((19|2[0-9])[0-9]{2})-(0[13578]|10|12)-(0[1-9]|[12][0-9]|3[01]))|(((19|2[0-9])[0-9]{2})-(0[469]|11)-(0[1-9]|[12][0-9]|30)))(Z|([+|-])([0-1][0-9]|2[0-3]):([0-5][0-9]))?"/>
```
Comment (david-waltermire-nist) [inline schema/xml/metaschema-datatypes.xsd:29]: Thinking about this more, we should use the same patterns as are used in dateTime.

```suggestion
			<xs:pattern value="(((2000|2400|2800|(19|2[0-9](0[48]|[2468][048]|[13579][26])))-02-29)|(((19|2[0-9])[0-9]{2})-02-(0[1-9]|1[0-9]|2[0-8]))|(((19|2[0-9])[0-9]{2})-(0[13578]|10|12)-(0[1-9]|[12][0-9]|3[01]))|(((19|2[0-9])[0-9]{2})-(0[469]|11)-(0[1-9]|[12][0-9]|30)))(Z|(-((0[0-9]|1[0-2]):00|0[39]:30)|\+((0[0-9]|1[0-4]):00|(0[34569]|10):30|(0[58]|12):45)))?"/>
```

File: schema/xml/metaschema-datatypes.xsd [hunk 2]
```diff
@@ -35,7 +35,7 @@
 			<xs:documentation>A string representing a 24-hour period with a required timezone.</xs:documentation>
 		</xs:annotation>
 		<xs:restriction base="DateDatatype">
-			<xs:pattern value="(((2000|2400|2800|(19|2[0-9](0[48]|[2468][048]|[13579][26])))-02-29)|(((19|2[0-9])[0-9]{2})-02-(0[1-9]|1[0-9]|2[0-8]))|(((19|2[0-9])[0-9]{2})-(0[13578]|10|12)-(0[1-9]|[12][0-9]|3[01]))|(((19|2[0-9])[0-9]{2})-(0[469]|11)-(0[1-9]|[12][0-9]|30)))(Z|[+-][0-9]{2}:[0-9]{2})"/>
+			<xs:pattern value="(((2000|2400|2800|(19|2[0-9](0[48]|[2468][048]|[13579][26])))-02-29)|(((19|2[0-9])[0-9]{2})-02-(0[1-9]|1[0-9]|2[0-8]))|(((19|2[0-9])[0-9]{2})-(0[13578]|10|12)-(0[1-9]|[12][0-9]|3[01]))|(((19|2[0-9])[0-9]{2})-(0[469]|11)-(0[1-9]|[12][0-9]|30)))(Z|([+|-])([0-1][0-9]|2[0-3]):([0-5][0-9]))"/>
```
Comment (david-waltermire-nist) [inline schema/xml/metaschema-datatypes.xsd:38]:
```suggestion
			<xs:pattern value="(((2000|2400|2800|(19|2[0-9](0[48]|[2468][048]|[13579][26])))-02-29)|(((19|2[0-9])[0-9]{2})-02-(0[1-9]|1[0-9]|2[0-8]))|(((19|2[0-9])[0-9]{2})-(0[13578]|10|12)-(0[1-9]|[12][0-9]|3[01]))|(((19|2[0-9])[0-9]{2})-(0[469]|11)-(0[1-9]|[12][0-9]|30)))(Z|(-((0[0-9]|1[0-2]):00|0[39]:30)|\+((0[0-9]|1[0-4]):00|(0[34569]|10):30|(0[58]|12):45)))"/>
```

**Comment by RS-Credentive** [comment]
Updated JSON to reflect the new timezones regex you recommended.

**Review by david-waltermire-nist — APPROVED** [review]
Looks good. Thank you for the additional edits.

**Review by david-waltermire-nist — APPROVED** [review]

**Pull request merged**